Bottom: 33c580ad45e8bd6e8bd28812b3c1ab8e9b3dfe96
Top:    fbda33be55a92f6aa7ed6296172ae7b366741ce9
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-04 22:57:38 +0000

Refresh of add-cache-for-board-outline

---

diff --git a/src/polygon.c b/src/polygon.c
index c7a591a..bb33330 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1639,20 +1639,21 @@ RestoreToPolygon (DataType * Data, int type, void *ptr1, void *ptr2)
 void
 ClearFromPolygon (DataType * Data, int type, void *ptr1, void *ptr2)
 {
-  LayerType *layer = (LayerType *) ptr1;
-
   if (!Data->polyClip)
     return;
 
   /* XXX: HACK, this is a kludgy place to put a check for recomputing our outline */
-  if (strcmp (layer->Name, "outline") == 0 ||
-      strcmp (layer->Name, "route") == 0)
+  if (type == LINE_TYPE || type == ARC_TYPE)
     {
-      Data->outline_valid = false;
+      LayerType *layer = (LayerType *) ptr1;
+
+      if (strcmp (layer->Name, "outline") == 0 ||
+          strcmp (layer->Name, "route") == 0)
+        Data->outline_valid = false;
     }
 
   if (type == POLYGON_TYPE)
-    InitClip (PCB->Data, layer, (PolygonType *) ptr2);
+    InitClip (PCB->Data, (LayerType *)ptr1, (PolygonType *) ptr2);
   else
     PlowsPolygon (Data, type, ptr1, ptr2, subtract_plow);
 }
